trillian/docs/ManualDeployment.md

```diff
@@ -0,0 +1,460 @@
+# CT Log Deployment (Manual)
+
+This document describes the individual steps and components involved in the
+deployment of a Trillian-based CT Log.  These steps will gradually build up a
+system as shown in the diagram below.
+
+<img src="images/DeploymentFull.png" width="650">
+
+Throughout the document, **cross-check**s are given to allow confirmation of
+successful setup.
+
+ - [Data Storage](#data-storage)
+ - [Trillian Services](#trillian-services)
+ - [Tree Provisioning](#tree-provisioning)
+ - [CT Personality](#ct-personality)
+    - [Key Generation](#key-generation)
+    - [CA Certificates](#ca-certificates)
+    - [CTFE Configuration](#ctfe-configuration)
+    - [CTFE Start-up](#ctfe-start-up)
+ - [Distribution](#distribution)
+ - [Primary Signer Election](#primary-signer-election)
+ - [Load Balancing](#load-balancing)
+ - [Monitoring](#monitoring)
+ - [Service Discovery](#service-discovery)
+ - [DoS Protection](#dos-protection)
+
+## Data Storage
+
+Data storage for the logged certificates is the heart of a CT Log.  The Trillian
+project has an internal storage interface that allows a variety of different
+implementations.
+
+This document uses the
+[MySQL storage implementation](https://github.com/google/trillian/blob/master/storage/mysql),
+which is set up according to the
+[instructions in the Trillian repo](https://github.com/google/trillian#mysql-setup);
+these instructions configure the Trillian database according to its
+[core schema file](http://github.com/google/trillian/blob/master//storage/mysql/storage.sql).
+
+
+**Cross-check**: At this point, manually connecting to the MySQL database should
+succeed:
+```
+% mysql --host=127.0.0.1 --port=3306 --user=root --database=test
+Welcome to the MariaDB monitor.  Commands end with ; or \g.
+Your MariaDB connection id is 764
+Server version: 10.1.29-MariaDB-6 Debian rodete
+
+Copyright (c) 2000, 2017, Oracle, MariaDB Corporation Ab and others.
+
+Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
+
+MariaDB [test]> show tables;
++-------------------+
+| Tables_in_test    |
++-------------------+
+| LeafData          |
+| MapHead           |
+| MapLeaf           |
+| SequencedLeafData |
+| Subtree           |
+| TreeControl       |
+| TreeHead          |
+| Trees             |
+| Unsequenced       |
++-------------------+
+9 rows in set (0.00 sec)
+
+MariaDB [test]> exit
+Bye
+```
+
+The setup so far is shown as:
+
+<img src="images/Deployment1DB.png" width="650">
+
+
+## Trillian Services
+
+The next step is to deploy two Trillian processes, the log server and the log
+signer.  These binaries are not specific to CT or to WebPKI certificates; they
+provide a general mechanism for transparently recording data in a Merkle tree.
+
+The log server (`github.com/google/trillian/server/trillian_log_server`) exposes
```

Comment:
Would it be a good idea to provide example full command lines?
As much as anything else to hint that now's the time to actually run these things up - since later text assumes that this has been done already.

Reply:
Added text at the start pointing to the shell scripts.